### PR TITLE
Update to syn 2.0

### PR DESCRIPTION
**Pull request by davidpdrsn**

### CODE REVIEW

**Comment by koute** [comment]
Thanks!

**Comment by davidpdrsn** [comment]
Okay. No rush from my side. We're already using our own fork.

File: speedy-derive/src/lib.rs [hunk 1]
```diff
@@ -531,25 +538,30 @@ fn is_c( attrs: &[syn::Attribute] ) -> bool {
 }
 
 fn parse_attributes< T >( attrs: &[syn::Attribute] ) -> Result< Vec< T >, syn::Error > where T: syn::parse::Parse {
-    struct RawAttributes< T >( syn::punctuated::Punctuated< T, Token![,] > );
+    struct RawAttributes< T >( Punctuated< T, Token![,] > );
 
     impl< T > syn::parse::Parse for RawAttributes< T > where T: syn::parse::Parse {
         fn parse( input: syn::parse::ParseStream ) -> syn::parse::Result< Self > {
+            dbg!(&input);
```
Comment (davidpdrsn) [inline speedy-derive/src/lib.rs:545]: @koute Sorry but I committed this by accident 😕 Should probably remove it before releasing.

Comment (koute) [reply speedy-derive/src/lib.rs:545]: Yeah, don't worry about it; I've already noticed it and removed it.

**Comment by koute** [comment]
I found the problem - on old syn the repr attributes are e.g. `(packed)` and on new syn they're just `packed`, which is why the tests are broken after the update.

**Comment by davidpdrsn** [comment]
> I found the problem - on old syn the repr attributes are e.g. `(packed)` and on new syn they're just `packed`, which is why the tests are broken after the update.

Yeah I noticed that as well. I'm gonna submit a fix :)

**Pull request merged**